modules/layouts/LayoutCSR.chpl

```diff
@@ -366,13 +366,25 @@ class CSRArr: BaseArr {
 
     // lookup the index and return the data or IRV
     const (found, loc) = dom.find(ind);
-    if setter && !found then
+    if !found then
       halt("attempting to assign a 'zero' value in a sparse array: ", ind);
     if found then
       return data(loc);
     else
       return irv;
   }
```

Comment:
Let's streamline the above dsiAccess:

if found then
  return data(loc);
else
  halt(...);